### PR TITLE
Set transactionNameSource to CUSTOM when setting transaction name

### PR DESCRIPTION
**Pull request by adinauer**
## :scroll: Description

Set source to CUSTOM when setting transaction name without source

## :bulb: Motivation and Context

Transaction name could be set by user and previously set source would be kept. 

## :green_heart: How did you test it?
Unit Test

## :pencil: Checklist

- [ ] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] No breaking changes

### CODE REVIEW

File: sentry/src/main/java/io/sentry/SentryTracer.java [hunk 1]
```diff
@@ -584,17 +584,17 @@ public void setMeasurement(
 
   @Override
   public void setName(@NotNull String name) {
-    if (root.isFinished()) {
-      return;
-    }
-
-    this.name = name;
+    setName(name, TransactionNameSource.CUSTOM);
```
Comment (marandaneto) [inline sentry/src/main/java/io/sentry/SentryTracer.java:587]: Do we call this method internally?
If we do, we cannot overwrite to `custom`, only if the user does.

**Review by romtsn — APPROVED** [review]

File: sentry/src/main/java/io/sentry/SentryTracer.java (hunk 1 above)
Comment (adinauer) [reply sentry/src/main/java/io/sentry/SentryTracer.java:587]: Good point will check. But if so we could simply change our code to use the other overload.

Comment (adinauer) [reply sentry/src/main/java/io/sentry/SentryTracer.java:587]: Couldn't find any direct calls to this overload (outside of tests).

Comment (adinauer) [reply sentry/src/main/java/io/sentry/SentryTracer.java:587]: Should we check cross platform SDKs before merging this?

Comment (adinauer) [reply sentry/src/main/java/io/sentry/SentryTracer.java:587]: @marandaneto should we check any SDKs first or can I go ahead and merge this PR?

Comment (marandaneto) [reply sentry/src/main/java/io/sentry/SentryTracer.java:587]: LGTM

**Pull request merged**